### PR TITLE
[feat] Implementing Support For Update Reset

### PR DESCRIPTION
**Pull request by ajisaxena**
This commit adds the support for update reset flow. Modifications to the verifier to check the reset reason and perform additional checks for the image being downloaded.

### CODE REVIEW

File: image/verify/src/lib.rs [hunk 1]
```diff
@@ -90,4 +90,13 @@ pub trait ImageVerificationEnv {
 
     // Get Device Lifecycle state
     fn dev_lifecycle(&self, image: Self::Image) -> Lifecycle;
+
+    // Get the vendor key index saved on cold boot in data vault
+    fn ven_pub_key_idx_dv(&self) -> u32;
+
+    // Get the owner key digest saved on cold boot in data vault
+    fn own_pub_key_digest_dv(&self) -> ImageDigest;
```
Comment (mhatrevi) [inline image/verify/src/lib.rs:98]: NIT: Spell out owner

File: image/verify/src/verifier.rs [hunk 2]
```diff
@@ -49,6 +49,10 @@ caliptra_err_def! {
         OwnerEccSignatureInvalidArg = 26,
         VendorPubKeyDigestInvalidArg = 27,
         VendorEccSignatureInvalidArg = 28,
+        UpdateResetOwnerDigestFailure=29,
```
Comment (mhatrevi) [inline image/verify/src/verifier.rs:52]: NIT: Spacing.

File: rom/dev/src/flow/update_reset.rs [hunk 3]
```diff
@@ -11,15 +11,35 @@ Abstract:
...
+    static mut MAN1_ORG: u8;
+}
+
+rom_err_def! {
+    UpdateReset,
+    UpdateResetErr
+    {
+        ManifestReadFailure = 0x2,
+        InavalidFirmwareCommand=0x03,
+        MailboxAccessFailure=0x04,
```
Comment (mhatrevi) [inline rom/dev/src/flow/update_reset.rs:33]: NIT: Inconsistent spacing.

File: rom/dev/src/flow/update_reset.rs [hunk 4]
```diff
@@ -11,15 +11,35 @@ Abstract:
...
+    static mut MAN2_ORG: u8;
+    static mut MAN1_ORG: u8;
+}
+
+rom_err_def! {
+    UpdateReset,
+    UpdateResetErr
+    {
+        ManifestReadFailure = 0x2,
+        InavalidFirmwareCommand=0x03,
```
Comment (mhatrevi) [inline rom/dev/src/flow/update_reset.rs:32]: Spelling: Invalid

File: rom/dev/src/flow/update_reset.rs [hunk 5]
```diff
@@ -28,10 +48,133 @@ impl UpdateResetFlow {
...
+            let ptr = &mut MAN2_ORG as *mut u8;
+
+            core::slice::from_raw_parts_mut(
+                ptr,
+                core::mem::size_of::<ImageManifest>()
+                    + manifest.fmc.size as usize
+                    + manifest.runtime.size as usize,
+            )
+        };
+        dst.clone_from_slice(src);
```
Comment (mhatrevi) [inline rom/dev/src/flow/update_reset.rs:127]: Old and new firmware sizes could be different.

File: rom/dev/src/verifier.rs [hunk 6]
```diff
@@ -102,4 +102,19 @@ impl<'a> ImageVerificationEnv for RomImageVerificationEnv<'a> {
...
+    fn ven_pub_key_idx_dv(&self) -> u32 {
+        self.env.data_vault().map(|dv| dv.vendor_pk_index())
+    }
+
+    /// Get the owner public key digest saved in the dv on cold boot
+    fn own_pub_key_digest_dv(&self) -> ImageDigest {
+        self.env.data_vault().map(|dv| dv.owner_pk_hash()).into()
+    }
+
+    // Save the fmc digest in the data vault on cold boot
```
Comment (mhatrevi) [inline rom/dev/src/verifier.rs:116]: Get

**Review by mhatrevi — CHANGES_REQUESTED** [review]
Added few comments to the PR; please take a look.

**Review by vsonims — APPROVED** [review]

**Review by mhatrevi — APPROVED** [review]

**Pull request merged**